cmip6_downscaling/methods/deepsd/tasks.py

```diff
@@ -0,0 +1,475 @@
+from dataclasses import asdict
+
+import numpy as np
+import tensorflow as tf
+import xarray as xr
+from carbonplan_data.metadata import get_cf_global_attrs
+from prefect import task
+from upath import UPath
+
+from ... import __version__ as version, config
+from ...data.observations import open_era5
+from ...data.utils import lon_to_180
+from ..common.bias_correction import bias_correct_gcm_by_method
+from ..common.containers import RunParameters, str_to_hash
+from ..common.utils import set_zarr_encoding, subset_dataset, zmetadata_exists
+from .utils import (
+    EPSILON,
+    INFERENCE_BATCH_SIZE,
+    bilinear_interpolate,
+    conservative_interpolate,
+    get_elevation_data,
+    initialize_empty_dataset,
+    normalize,
+    output_node_name,
+    res_to_str,
+    stacked_model_path,
+    starting_resolutions,
+)
+
+scratch_dir = UPath(config.get("storage.scratch.uri"))
+intermediate_dir = UPath(config.get("storage.intermediate.uri")) / version
+results_dir = UPath(config.get("storage.results.uri")) / version
+use_cache = config.get('run_options.use_cache')
+
+xr.set_options(keep_attrs=True)
+
+
+@task(log_stdout=True)
+def shift(path: UPath, path_type: str, run_parameters: RunParameters) -> UPath:
+    """Interpolate obs data to grid specs in ``xe.util.grid_global``.
+
+    Parameters
+    ----------
+    path : UPath
+        Path to original dataset.
+    path_type : str
+        Specify whether gcm or obs to set output resolution
+    run_parameters : RunParameters
+        Parameters for run set-up and model specs
+
+    Returns
+    -------
+    UPath
+        Path to shifted dataset.
+    """
+    if path_type == 'obs':
+        output_degree = 0.25
+    elif path_type == 'gcm':
+        output_degree = starting_resolutions[run_parameters.model]
+    else:
+        raise ValueError('path_type must be gcm or obs')
+
+    ds_hash = str_to_hash(str(path) + str(output_degree))
+    target = intermediate_dir / 'shift' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        shifted_ds = xr.open_zarr(target)
+        return target
+
+    orig_ds = xr.open_zarr(path)
+
+    # Note: Notebook included unit conversion at this step for precipitation, but this is done earlier in `load_cmip`
+    shifted_ds = bilinear_interpolate(ds=orig_ds, output_degree=output_degree)
+    shifted_ds.attrs.update({'title': 'shift'}, **get_cf_global_attrs(version=version))
+    print(f'writing shifted dataset to {target}')
+    shifted_ds.pipe(set_zarr_encoding).to_zarr(target, mode='w')
```

Comment:
rather than calling `.to_zarr(...)`,could we use the `blocking_to_zarr(...)` utility function to write datasets to zarr  for consistency and zarr store validation? https://github.com/carbonplan/cmip6-downscaling/blob/3c30452edf7f7f8dc755e7f9a3090126d9885761/cmip6_downscaling/methods/common/utils.py#L65

Reply:
Thanks for the review, @andersy005! I'll add these changes.